app/models/category.py

```diff
@@ -0,0 +1,101 @@
+from sqlalchemy import String, Boolean, Integer, Text, Date, DECIMAL, ForeignKey, Enum
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from uuid import UUID
+from .base import Base
+from .evaluation import ComprehensiveEvaluation
+from .user import User
+from enum import Enum as PyEnum
+from .evaluation import RecommendedChallenge
+
+
+#  CategoryNameEnum 클래스 정의
+class CategoryNameEnum(PyEnum):
+    CODING = "코딩"
+    ENGLISH = "영어"
+    WORKOUT = "운동"
+
+
+# CategoryColorEnum 클래스 정의
+class CategoryColorEnum(PyEnum):
+    CODING = "#FDA63A"  # 호박색
+    ENGLISH = "#6C88C4"  # 인디고 블루
+    WORKOUT = "#556B2F"  # 올리브 그린
+
+
+# CategoryProgress 테이블 (한 카테고리에 여러 목표, 노트가 연결됨)
+class CategoryProgress(Base):
+    __tablename__ = "category_progresses"
+
+    user_id: Mapped[UUID] = mapped_column(
+        UUID(as_uuid=True), ForeignKey("users.id"), nullable=False
+    )
+    category_name: Mapped[CategoryNameEnum] = mapped_column(
+        Enum(CategoryNameEnum, name="category_name_enums"), nullable=False
+    )
+
+    total_goal: Mapped[int] = mapped_column(Integer, nullable=False)
+    completed_goal: Mapped[int] = mapped_column(Integer, nullable=False)
+    progress_rate: Mapped[DECIMAL] = mapped_column(DECIMAL(5, 2), default=0.00)
+
+    # CategoryProgress → User (N : 1 관계)
+    users: Mapped["User"] = relationship(
+        "User", back_populates="category_progresses", cascade="all, delete"
+    )
+
+    # CategoryProgress → ComprehensiveEvaluation, RecommendedChallenge, Goal (1 : N 관계)
+    comprehensive_evaluations: Mapped[list["ComprehensiveEvaluation"]] = relationship(
```

Comment:
```suggestion
    comprehensive_evaluations: Mapped[List["ComprehensiveEvaluation"]] = relationship(
```
@Ryan-OH 
조금 뒷북같지만 죄송합니다 ㅠㅠ sqlalchemy 공식문서 좀 보다가 발견했는데 관계에서 list인 부분들 전체적으로 typing 의 List로 안해도 문제 없을까요?

링크: https://docs.sqlalchemy.org/en/20/orm/quickstart.html#declare-models
<img width="857" alt="스크린샷 2025-04-06 오후 2 44 46" src="https://github.com/user-attachments/assets/1488798c-65ce-4dda-bb16-90ff15483fb9" />

Reply:
기존 코드 (list[...]) → 작동 가능하지만, IDE 자동완성 및 타입 검사에서 조금 덜 명확할 수 있음
 개선 코드 (List[...]) → IDE 및 타입 검사에서 더 명확하게 인식되므로 코드 가독성이 좋아짐
바꾸는 쪽이 더 좋은것 같긴합니다 바꿀까요?